src/transaction/log_applier.c

```diff
@@ -7043,141 +7043,174 @@ la_print_log_arv_header (const char *database_name, LOG_ARV_HEADER * hdr, bool v
-      printf ("%-30s : %ld\n", "Delete count", ha_apply_info.delete_counter);
-      printf ("%-30s : %ld\n", "Schema count", ha_apply_info.schema_counter);
-      printf ("%-30s : %ld\n", "Commit count", ha_apply_info.commit_counter);
-      printf ("%-30s : %ld\n", "Fail count", ha_apply_info.fail_counter);
+  if (verbose)
+    {
+      db_datetime_to_string ((char *) timebuf, 1024, &ha_apply_info.start_time);
+      printf ("%-30s : %s\n", "Start time", timebuf);
+    }
 
-      if (verbose)
+  if (check_replica_info)
+    {
+      replica_time_bound_str = prm_get_string_value (PRM_ID_HA_REPLICA_TIME_BOUND);
+      db_datetime_to_string2 ((char *) timebuf, 1024, &ha_apply_info.log_record_time);
+
+      printf ("\n *** Replica-specific Info. *** \n");
+      if (replica_time_bound_str == NULL)
 	{
-	  db_datetime_to_string ((char *) timebuf, 1024, &ha_apply_info.start_time);
-	  printf ("%-30s : %s\n", "Start time", timebuf);
+	  printf ("%-30s : %d second(s)\n", "Deliberate lag", prm_get_integer_value (PRM_ID_HA_REPLICA_DELAY_IN_SECS));
 	}
 
-      if (check_replica_info)
+      printf ("%-30s : %s\n", "Last applied log record time", timebuf);
+      if (replica_time_bound_str != NULL)
 	{
-	  replica_time_bound_str = prm_get_string_value (PRM_ID_HA_REPLICA_TIME_BOUND);
-	  db_datetime_to_string2 ((char *) timebuf, 1024, &ha_apply_info.log_record_time);
-
-	  printf ("\n *** Replica-specific Info. *** \n");
-	  if (replica_time_bound_str == NULL)
-	    {
-	      printf ("%-30s : %d second(s)\n", "Deliberate lag",
-		      prm_get_integer_value (PRM_ID_HA_REPLICA_DELAY_IN_SECS));
-	    }
-
-	  printf ("%-30s : %s\n", "Last applied log record time", timebuf);
-	  if (replica_time_bound_str != NULL)
-	    {
-	      printf ("%-30s : %s\n", "Will apply log records up to", replica_time_bound_str);
-	    }
+	  printf ("%-30s : %s\n", "Will apply log records up to", replica_time_bound_str);
 	}
     }
+
 check_applied_info_end:
-  if (error != NO_ERROR)
+
+  return error;
+}
+
+/*
+ * la_get_copied_log_info() - when applyinfo utility is executed with -L or -p option, this fuction will try to get copied log or copied log page information.
+ *   return: NO_ERROR or error code
+ *   database_name(in): db name
+ *   log_path(in): real path of log file
+ *   page_num(in): the number of log page 
+ *   verbose(in): applyinfo -v option
+ *   copied_eof_lsa(in/out): inforamtion for delayed log count
+ *   copied_append_lsa(in/out): current append location
+ */
+int
+la_get_copied_log_info (const char *database_name, const char *log_path, INT64 page_num, bool verbose,
+			LOG_LSA * copied_eof_lsa, LOG_LSA * copied_append_lsa)
+{
+  int error = NO_ERROR;
+  char active_log_path[PATH_MAX];
+  char log_path_buf[PATH_MAX];
+
+  assert (database_name != NULL);
+  assert (log_path != NULL);
+
+  if (realpath (log_path, log_path_buf) != NULL)
     {
-      printf ("%s\n", db_error_string (3));
```

Comment:
Any other errors than the `res < 0` case in `la_get_applied_log_info() is not printed?

Reply:
1. db_datetime_to_string() do not execute er_set().
2. db_datetime_to_string2()
        db_datetime_decode() : do not execute er_set()
3. snprintf() : do not execute er_set()

Reply:
1. db_datetime_to_string()
2. db_datetime_to_string2()
do not set er_set()

In the db_datetime_to_string2() function, snprintf() returns error as negative value but not execute er_set(). Original code didn't handle it. is this bug?